### PR TITLE
kubeedge: refactored the folder structure

### LINKED ISSUE #159: Realign code structure to incorporate cloud part
**What would you like to be added**:
The code structuring needs to be changed in order to add the cloud part modules.

**Why is this needed**:
Code modularity will help in easy maintenance once the cloud part is opensourced.
This will help reusing common framework like beehive , maintaing a common vendor package for both cloud and edge modules.

### PR DESCRIPTION
**Pull request by sids-b**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
refactored the folder structure of kubeedge to accomodate both edge and
cloud part
Fixes #159 

**Special notes for your reviewer**:
Request to review all files and their placement so as to avoid refactoring later. 
Tested with IEF and its working fine :)

### CODE REVIEW

**Comment by rohitsardesai83** [comment]
This PR fixes #159

File: Makefile [hunk 1]
```diff
@@ -1,30 +1,18 @@
...
+.PHONY: edge_verify
+edge_verify:
+	cd edge && $(MAKE) verify
 
-.PHONY: clean
-clean:
-	bash -x hack/clean.sh
+.PHONY: edge_integration_test
+edge_integration_test:
+	cd edge && $(MAKE) integration_test
```
Comment (skdwriting) [inline Makefile]: Later : Please consider Common tests (usually all e2e or functional tests will involve both cloud and edge)

File: README.md [hunk 2]
```diff
@@ -122,17 +122,17 @@ yum install mosquitto
 
 See [mosquitto official website](https://mosquitto.org/download/) for more information.
 
-### Build
+### Build Edge
 
 Clone kube-edge
```
Comment (skdwriting) [inline README.md]: Please make the names consistent (low prio) : kubeedge / KubeEdge

File: edge/pkg/devicetwin/dtcontext/dtcontext.go [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-	"github.com/kubeedge/kubeedge/beehive/pkg/common/config"
-	"github.com/kubeedge/kubeedge/beehive/pkg/common/log"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core/context"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/common/config"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/model"
```
Comment (kevin-wangzefeng) [inline edge/pkg/devicetwin/dtcontext/dtcontext.go]: please move the github.com/* section together with the following lines that importing github.com/*

File: edge/pkg/devicetwin/dtmanager/membership.go [hunk 4]
```diff
@@ -1,20 +1,20 @@
 package dtmanager
 
 import (
-	"github.com/kubeedge/kubeedge/beehive/pkg/common/log"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/model"
```
Comment (kevin-wangzefeng) [inline edge/pkg/devicetwin/dtmanager/membership.go]: please move the github.com/* section together with the following lines that importing github.com/*

File: edge/pkg/devicetwin/dtmanager/twin.go [hunk 5]
```diff
@@ -1,20 +1,20 @@
 package dtmanager
 
 import (
-	"github.com/kubeedge/kubeedge/beehive/pkg/common/log"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/model"
```
Comment (kevin-wangzefeng) [inline edge/pkg/devicetwin/dtmanager/twin.go]: please move the github.com/* section together with the following lines that importing github.com/*

File: edge/pkg/edged/status/status_manager.go [hunk 6]
```diff
@@ -3,17 +3,17 @@ package status
 import (
 	"time"
 
-	"github.com/kubeedge/kubeedge/pkg/edged/podmanager"
+	"github.com/kubeedge/kubeedge/edge/pkg/edged/podmanager"
```
Comment (kevin-wangzefeng) [inline edge/pkg/edged/status/status_manager.go]: please move this line together with the github.com/kubeedge/kubedge/* import section

File: edge/pkg/edgehub/clients/factory.go [hunk 7]
```diff
@@ -1,10 +1,10 @@
 package clients
 
 import (
-	"github.com/kubeedge/kubeedge/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/common/log"
```
Comment (kevin-wangzefeng) [inline edge/pkg/edgehub/clients/factory.go]: redundant empty line

File: edge/pkg/edgehub/common/http/http_test.go [hunk 8]
```diff
@@ -26,8 +26,8 @@ import (
 	"strconv"
 	"testing"
 
-	"github.com/kubeedge/kubeedge/pkg/common/util"
-	common "github.com/kubeedge/kubeedge/pkg/edgehub/common"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/util"
+	common "github.com/kubeedge/kubeedge/edge/pkg/edgehub/common"
```
Comment (kevin-wangzefeng) [inline edge/pkg/edgehub/common/http/http_test.go]: please move this section after line: "github.com/stretchr/testify/assert"

File: edge/pkg/edgehub/controller.go [hunk 9]
```diff
@@ -7,15 +7,15 @@ import (
...
-
-	"github.com/kubeedge/kubeedge/pkg/common/message"
-	"github.com/kubeedge/kubeedge/pkg/edgehub/clients"
-	http_utils "github.com/kubeedge/kubeedge/pkg/edgehub/common/http"
-	"github.com/kubeedge/kubeedge/pkg/edgehub/config"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/model"
+
```
Comment (kevin-wangzefeng) [inline edge/pkg/edgehub/controller.go]: redundant empty line

File: edge/cmd/edge_core.go [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/dbm"
+	_ "github.com/kubeedge/kubeedge/edge/pkg/devicetwin"
+	_ "github.com/kubeedge/kubeedge/edge/pkg/edged"
+	_ "github.com/kubeedge/kubeedge/edge/pkg/edgehub"
+	_ "github.com/kubeedge/kubeedge/edge/pkg/eventbus"
+	_ "github.com/kubeedge/kubeedge/edge/pkg/metamanager"
+	_ "github.com/kubeedge/kubeedge/edge/test"
+	// _ "github.com/kubeedge/kubeedge/pkg/edgefunction"
```
Comment (kevin-wangzefeng) [inline edge/cmd/edge_core.go]: can you remove this line?

File: edge/pkg/metamanager/module.go [hunk 11]
```diff
@@ -3,13 +3,13 @@ package metamanager
...
 	"time"
 
-	"github.com/kubeedge/kubeedge/beehive/pkg/common/config"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core/context"
-	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/common/config"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/common/beehive/pkg/core/model"
```
Comment (kevin-wangzefeng) [inline edge/pkg/metamanager/module.go]: redundant emptyline

File: edge/cmd/edge_core.go (hunk 10 above)
Comment (sids-b) [reply edge/cmd/edge_core.go]: fixed

File: edge/pkg/edgehub/controller.go (hunk 9 above)
Comment (sids-b) [reply edge/pkg/edgehub/controller.go]: fixed

File: edge/pkg/edgehub/clients/factory.go (hunk 7 above)
Comment (sids-b) [reply edge/pkg/edgehub/clients/factory.go]: fixed

File: edge/pkg/edgehub/common/http/http_test.go (hunk 8 above)
Comment (sids-b) [reply edge/pkg/edgehub/common/http/http_test.go]: fixed

File: edge/pkg/edged/status/status_manager.go (hunk 6 above)
Comment (sids-b) [reply edge/pkg/edged/status/status_manager.go]: fixed

File: edge/pkg/devicetwin/dtmanager/twin.go (hunk 5 above)
Comment (sids-b) [reply edge/pkg/devicetwin/dtmanager/twin.go]: fixed

File: edge/pkg/devicetwin/dtcontext/dtcontext.go (hunk 3 above)
Comment (sids-b) [reply edge/pkg/devicetwin/dtcontext/dtcontext.go]: fixed

File: README.md (hunk 2 above)
Comment (sids-b) [reply README.md]: Fixed

**Comment by kevin-wangzefeng** [comment]
There are still some importing order issue, but not blocking.
Let's get fix them in another cleanup PR.

**Comment by kevin-wangzefeng** [comment]
LGTM

**Pull request merged**